test/tests/jobs/GetJobTest.cpp

```diff
@@ -699,5 +700,104 @@ struct GetJobTest : tpunit::TestFixture {
         command["jobPriority"] = "111";
         tester->executeWaitVerifyContent(command, "402 Invalid priority value");
     }
+
+    void testRetryableParentJobs() {
+        // Create the parent job
+        SData createJobCommand("CreateJob");
+        createJobCommand["name"] = "ParentJob";
+        createJobCommand["retryAfter"] = "+2 SECONDS";
+        string parentJobID = tester->executeWaitVerifyContentTable(createJobCommand)["jobID"];
+        ASSERT_GREATER_THAN(stol(parentJobID), 0);
+
+        // Get it
+        SData getJobCommand("GetJob");
+        getJobCommand["name"] = "ParentJob";
+        STable getJobResponse = tester->executeWaitVerifyContentTable(getJobCommand);
+
+        // Verify we have a RUNQUEUED job
+        ASSERT_EQUAL(getJobResponse["jobID"], parentJobID);
+        string state = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(parentJobID) + ";");
+        ASSERT_EQUAL(state, "RUNQUEUED");
+
+        // Create a child
+        createJobCommand.clear();
+        createJobCommand.methodLine = "CreateJob";
+        createJobCommand["name"] = "ChildJob";
+        createJobCommand["parentJobID"] = parentJobID;
+        string childJobID = tester->executeWaitVerifyContentTable(createJobCommand)["jobID"];
+        string childState = tester->readDB("SELECT state FROM jobs WHERE jobID=" + SQ(childJobID) + ";");
+        ASSERT_EQUAL(childState, "PAUSED");
+
+        // Wait a bit for retryAfter to kick in
+        sleep(3);
```

Comment:
Everywhere we do this eventually causes travis failures, so I like to put the check in a loop that runs until the condition is correct or some fairly long (1 minute?) timeout occurs.

Reply:
I disagree, because:
1. We do this above in several places
2. I think this makes sense. After 3 seconds, we should see the job being ready to be retried since its retryAfter is 2 seconds. If not, there's probably a bug, right?